### ISSUE TITLE
Compatibility with rspec 3.5

### ISSUE THREAD

**Issue by dbackeus** [mongoid-rspec/mongoid-rspec#174]
Running with 3.5.x gives deprecation warning:
```
Mongoid::Matchers::Validations::ValidateUniquenessOfMatcher implements a legacy RSpec matcher
protocol. For the current protocol you should expose the failure messages
via the `failure_message` and `failure_message_when_negated` methods.
```

**Comment by narutosanjiv** [comment]
@dbackeus Can you still getting this compatibility warning?. I have tried to repro on my dev environment but failed to do so.

**Comment by dbackeus** [comment]
Sorry I just noticed that mongoid-rspec had downgraded to 1.x after Mongoid 6.0.0.rc0 upgrade (since mongoid-rspec 3.0.0 is locked to Mongoid 5). So the actual issue here is likely the mongoid compatibility...

**Comment by bgetting** [comment]
```
Bundler could not find compatible versions for gem "mongoid":
  In Gemfile:
    mongoid (~> 6.0.0)

    mongoid-rspec (~> 3.0) was resolved to 3.0.0, which depends on
      mongoid (~> 5.0)
```
